governance/delegation-interop.md

```diff
@@ -0,0 +1,65 @@
+# Purpose
+
+The primary purpose of this design document is to give an approach to support voting power delegation across multiple L2's that employ the `GovernanceDelegation` and `GovToken` as a
+pre-deploy. It will give context around the desired behavior of the interactions between the contracts that exist on both OP mainnet as well as each of these bespoke chains. Furthermore, this document serves as an exploration into a few choice mechanisms that allow users the ability to do partial delegation with tokens they posses on each of the respective chains.
+
+# Summary
+
+There will be two main changes to the existing `GovernanceDelegation` contract to support interoperability. First, the function in which the checkpoints can be queried from the
+contract will now check for the migration status prior to retrieving voting power and implement an announce function that will emit an event that contains the chain ID and the block
+number of the latest checkpoint. The contract on OP mainnet will then consume these events, and in its own storage provide a mapping between chainIds and an object containing both
+the latest checkpoint and the block number. After tokens are transferred using the `GovToken` contract a check is performed on the delegation contract that ensures that before this
+mapping is updated, the nonce of the token transfer received has been incremented as well as being sent from a block number greater than the current one.
+
+# Problem Statement + Context
+
+In the existing `GovernanceDelegation` and `GovToken` contract it does not support interoperability, thus making the voting power fragmented across all of the potential L2 chains. The negative consequence of this is that tracking delegations becomes a much harder task when users have partially delegated to many different users that may or may not be on the same chain.
+
+However, pushing all of the logic onto the `GovToken` proves to be problematic given the potential flood of messages that may be out of sequence or have many smaller transfer events
+prior to a single delegation event. The following section aims to outline a solution to correct this as well as ensure that relayers still have the proper incentives to handle these
```

Comment:
Can you expand on this problem? I don't follow the "problematic given the potential flood of messages that may be out of sequence or have many smaller transfer events" aspect

Reply:
The goal here was to provide some context around the previous solution which had a hook `afterTokenTransfer` in which a message was sent between L2's to OP main net that then updated the checkpoints. This is no longer pertinent to the design but was trying to lay the groundwork of some prior thinking around supporting this feature. Here's the PR of the old spec that it looks like we are moving away from: https://github.com/ethereum-optimism/specs/pull/285